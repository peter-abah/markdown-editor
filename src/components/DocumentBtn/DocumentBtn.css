.doc-btn button {
  display: flex;
  flex-direction: column;
  width: 100%;
  margin-bottom: 1rem;
  padding: 0.5rem;
}

.doc-btn__date {
  font-weight: 400;
  color: var(--c-light-gray);
}

.doc-btn__name: {
  color: var(--c-orange);
}

.doc-btn:last {
  margin-bottom: 0;
}