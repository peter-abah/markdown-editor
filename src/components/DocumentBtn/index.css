.doc-btn button {
  display: flex;
  flex-direction: column;
  margin-bottom: 1rem;
  padding: 0.5rem;
}

.doc-btn__date {
  font-weight: 400;
  color: #aaa;
}

.doc-btn__name: {
  color: var(--color-orange);
}

.doc-btn:last {
  margin-bottom: 0;
}