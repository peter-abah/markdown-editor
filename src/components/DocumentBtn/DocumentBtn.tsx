import React from 'react';
import clsx from 'clsx';
import { Document } from '@/types';
import { MdOutlineDescription as FileIcon } from 'react-icons/md';
import { parseISO, format } from 'date-fns';
import './DocumentBtn.css';

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  doc: Document;
  onClick: () => void;
  isActive?: boolean;
};
const DocumentBtn = ({doc, isActive, ...btnProps}: Props) => {
  const formattedDate = format(parseISO(doc.updated_at), 'MMMM d, yyyy');
  return (
    <li >
      <button
        className={clsx("doc-btn", isActive && 'active')}
        {...btnProps}
      >
        <FileIcon className='file-icon mr-2' />
        <div className='flex flex-col items-start'>
          <span className="doc-btn__date">{formattedDate}</span>
          <span className="doc-btn__name">{doc.name}</span>
        </div>
      </button>
    </li>
  )
};

export default DocumentBtn;