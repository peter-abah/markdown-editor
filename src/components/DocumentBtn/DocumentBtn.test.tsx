import React from 'react';
import userEvent from '@testing-library/user-event';
import {render, fireEvent, waitFor, screen} from '@testing-library/react';

import DocumentBtn from '.';

const doc = {
  id: 'first',
  name: 'render',
  content: '',
  created_at: new Date(),
  updated_at: new Date(),
};

test('it renders correctly', () => {
  const {container} = render(
    <DocumentBtn
      doc={doc}
      onClick={() => null}
    />
  );
  expect(container).toMatchSnapshot();
});

test('it renders with document name and date', () => {
  // Using epoch date as date (Jan 1 1970) for updated at since
  // That is what should be displayed
  const docWithEpochDate = {
    ...doc,
    created_t: new Date(),
    updated_at: new Date(0),
  };
  
  const formattedDate = "January 1, 1970"
  
  const { getByRole } = render(
    <DocumentBtn
      doc={docWithEpochDate}
      onClick={() => null}
    />
  );
  
  const btn = getByRole('button');
  expect(btn).toHaveTextContent(docWithEpochDate.name);
  expect(btn).toHaveTextContent(formattedDate);
});

test("it responds to click events", async () => {
  const mockFn = jest.fn();
  
  const { getByRole } = render(
    <DocumentBtn
      doc={doc}
      onClick={mockFn}
    />
  );
  
  const btn = getByRole('button');
  await userEvent.click(btn);
  
  expect(mockFn).toHaveBeenCalled();
});