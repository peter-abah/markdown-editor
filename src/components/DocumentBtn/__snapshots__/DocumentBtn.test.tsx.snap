// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <li
    class="doc-btn"
  >
    <button>
      <span
        class="doc-btn__date"
      >
        June 21, 2022
      </span>
      <span
        class="doc-btn__name"
      >
        render
      </span>
    </button>
  </li>
</div>
`;
